File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int, int> count = new Dictionary<int, int>();
        // nums = [1,1,2,2,3] = 5
        //List<int>[5] => the zero cannot be calculated
        List<int>[] freq = new List<int>[nums.Length + 1];

        for(int i = 0; i < freq.Length; i++) {
            freq[i] = new List<int>();
        }

        foreach(int num in nums) {
            if(count.ContainsKey(num)) {
                count[num]++;
            } else {
                count[num] = 1;
            }
        }

        foreach(var entry in count) {
            freq[entry.Value].Add(entry.Key);
        }

        //freq[2] = [1, 2]

        int[] result = new int[k];
        int index = 0;

        for(int i = freq.Length - 1; i > 0 && index < k; i--) {
            foreach(int n in freq[i]) {
                result[index++] = n;
                if(index == k) {
                    return result;
                }
            }
        }
        return result;

    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-10.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        n = len(nums)
        maxSum = 0

        for num in nums:
            maxSum += num
        
        if maxSum % 2 != 0:
            return False
        
        def dfs(i, currTotal):
            if i == n or currTotal > (maxSum / 2):
                return False
            if currTotal == (maxSum / 2):
                return True
            
            return dfs(i + 1, currTotal) or dfs(i + 1, currTotal + nums[i])
            
        return dfs(0,0)
            

